### DISCUSSION TITLE
Restoring data

### DISCUSSION THREAD

**Discussion by Tthongsy** [datalust/seq-tickets#1309] [Help]
Hey,

I was wondering if there was a way to restore data that has aged out of the system or past the retention policies?

Thanks,
Tommy T

**Comment by nblumhardt** [comment]
Hello!

Once data has aged out/been deleted from the event store, it's not possible to restore it.

However, if you set up an instance of _Seq.App.JsonArchive_, and choose _Stream incoming events_, this will write all event data to a set of newline-delimited JSON files in a location of your choice (you can mount an enormous SAN drive for this, for example).

Those newline-delimited files will be in the CLEF format, which `seqcli ingest --json -i <file path or glob>` can read, allowing the events to be selectively imported back into the same Seq server (or another one) for analysis when required.

If you need more info on any of the above, please just let me know. Hope this helps!

Nick

**Reply by Tthongsy** [reply]
Out of curiosity of recommendation, instead of that, would it be good if I just restored the entire Seq instance in a different VM and connect to that machine or copying the files onto another instance of Seq? Would these methods be ok to use as methods of restoring data, since we backup the seq servers?

**Reply by nblumhardt** [reply]
Copying the `Stream/` folder from one machine to another will work, as long as both Seq instances are stopped when the folder is copied out/in, and the destination folder is completely deleted/replaced.

Restoring an entire VM snapshot should also work.

Copying the `Seq` folder won't work reliably, since on Windows the data relies on machine-specific DPAPI encryption. Backup + restore (using Seq's backup/restore) mechanism is the best way to get the all of the configuration data between machines.
